cli/command/service.go

```diff
@@ -10,6 +16,260 @@ func NewServiceCommand(b *CommandBase) *ServiceCommand {
+			return err
+		}
+
+		return s.printer.PrintServices(service)
+	}
+
+	return s.waitOnJobHelper(c, jobID, "creating", onCompleteFN)
+}
+
+func (s *ServiceCommand) delete(c *cli.Context) error {
+	deleteFN := func(serviceID string) (string, error) {
+		return s.client.DeleteService(serviceID)
+	}
+
+	return s.deleteHelper(c, "service", deleteFN)
+}
+
+func (s *ServiceCommand) list(c *cli.Context) error {
+	serviceSummaries, err := s.client.ListServices()
+	if err != nil {
+		return err
+	}
+
+	return s.printer.PrintServiceSummaries(serviceSummaries...)
+}
+
+func (s *ServiceCommand) logs(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME")
+	if err != nil {
+		return err
+	}
+
+	serviceID, err := s.resolveSingleEntityIDHelper("service", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	query := buildLogQueryHelper(serviceID, c.String("start"), c.String("end"), c.Int("tail"))
+
+	logs, err := s.client.ReadServiceLogs(serviceID, query)
+	if err != nil {
+		return err
+	}
+
+	return s.printer.PrintLogs(logs...)
+}
+
+func (s *ServiceCommand) read(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME")
+	if err != nil {
+		return err
+	}
+
+	serviceIDs, err := s.resolver.Resolve("service", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	services := make([]*models.Service, len(serviceIDs))
+	for i, serviceID := range serviceIDs {
+		service, err := s.client.ReadService(serviceID)
+		if err != nil {
+			return err
+		}
+
+		services[i] = service
+	}
+
+	return s.printer.PrintServices(services...)
+}
+
+func (s *ServiceCommand) scale(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME", "COUNT")
+	if err != nil {
+		return err
+	}
+
+	serviceID, err := s.resolveSingleEntityIDHelper("service", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	scale, err := strconv.Atoi(args["COUNT"])
+	if err != nil {
+		return fmt.Errorf("Failed to parse COUNT argument: %v", err)
+	}
+
+	req := models.UpdateServiceRequest{
+		ServiceID: serviceID,
+		Scale:     &scale,
+	}
+
+	jobID, err := s.client.UpdateService(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFN := func(serviceID string) error {
+		service, err := s.client.ReadService(serviceID)
+		if err != nil {
+			return err
+		}
+
+		return s.printer.PrintServices(service)
+	}
+
+	return s.waitOnJobHelper(c, jobID, "scaling", onCompleteFN)
+}
+
+func (s *ServiceCommand) update(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME", "DEPLOY")
+	if err != nil {
+		return err
+	}
+
+	serviceID, err := s.resolveSingleEntityIDHelper("service", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := s.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.UpdateServiceRequest{
+		ServiceID: serviceID,
+		DeployID:  &deployID,
+	}
+
+	jobID, err := s.client.UpdateService(req)
+	if err != nil {
+		return err
+	}
+
+	onCompleteFN := func(serviceID string) error {
```

Comment:
We need to add back in the `WaitForDeployment` helper function from develop and do that before returning

Reply:
✅